src/bindings/node/node.cc

```diff
@@ -26,52 +28,71 @@ boost::property_tree::ptree make_conf(const char* config) {
   return json_to_pt(config);
 }
 
-// check if napi_status is napi_ok, throw error if not
-void checkNapiStatus(napi_status status, napi_env env, const char* error_message) {
-  if (status != napi_ok) {
-    napi_throw_error(env, NULL, error_message);
+class ActorWorker : public Napi::AsyncWorker {
+public:
+  ActorWorker(Napi::Function& callback,
+              const std::string request,
+              valhalla::tyr::actor_t& actor,
+              const std::function<std::string(valhalla::tyr::actor_t& actor,
+                                              const std::string& request)>& func)
+      : Napi::AsyncWorker(callback), request(request), actor(actor), func(func) {
+  }
+  ~ActorWorker() {
+  }
+
+  void Execute() {
+    try {
+      response = func(actor, request);
+    } catch (const valhalla::valhalla_exception_t& e) {
+      auto http_code = ERROR_TO_STATUS.find(e.code)->second;
+      std::string err_message = "{ \"error_code\": " + std::to_string(e.code) +
+                                ", \"http_code\": " + std::to_string(http_code) +
+                                ", \"message\": \"" + e.message + "\" }";
```

Comment:
Is there any risk of `e.message` containing double quotes and breaking the JSON?

Reply:
its very unlikely we control basically all of the messages: https://github.com/valhalla/valhalla/blob/master/valhalla/exception.h

in very rare cases you might get an exception from something we didnt plan for and somehow didnt also `catch` like `std::bad_alloc` or maybe if you are writing some new code `string_base::at` you know standard exception that are the result of bugs. again though most of these we catch and throw our more generic failure message because we dont want users to get some stl generated exception message.